rules/crowdstrike_rules/event_stream_rules/crowdstrike_single_ip_allowlisted.py

```diff
@@ -0,0 +1,64 @@
+from crowdstrike_event_streams_helpers import audit_keys_dict, cs_alert_context
+
+
+def get_single_ips(event, fieldname="cidrs") -> list[str]:
+    """Searches the "cidrs" field of the event audit keys, and returns any cidr entries which
+    are actually just single IP addresses."""
+    single_ips = []
+    audit_keys = audit_keys_dict(event)
+    cidrs = str_to_list(audit_keys[fieldname])
+    for entry in cidrs:
+        if "/" not in entry:
+            single_ips.append(entry)
+        elif entry.endswith("/32"):
+            # A 32-bit CIDR range is the same as a single IP address
+            single_ips.append(entry[:-3])
+    return single_ips
+
+
+def str_to_list(liststr: str) -> list[str]:
```

Comment:
If this pattern shows up in other rules it may be worth moving this to the crowdstrike helpers

Reply:
You're right - I think this will probably be useful in future rules so I'll move it to the helper module!